src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -121,6 +169,21 @@ public override bool TrySetPropertyValue(string name, object value)
                 throw Error.ArgumentNull("name");
             }
 
+            if (_instanceAnnotationsPropertyInfo != null)
+            {                
+                if (name == _instanceAnnotationsPropertyInfo.Name)
+                {
+                    IODataInstanceAnnotationContainer annotationValue = value as IODataInstanceAnnotationContainer;
+                    if (_instanceAnnotationCache == null && annotationValue != null)
```

Comment:
>_instanceAnnotationCache == null [](start = 24, length = 32)

Do we not allow them to reset the value of the _instanceAnnotationCache?

Reply:
Yes, like for dynamicdictionary